tests/integrationv2/test_signature_algorithms.py

```diff
@@ -74,8 +78,13 @@ def test_s2n_server_signature_algorithms(managed_process, cipher, provider, prot
         use_client_auth=client_auth,
         key=certificate.key,
         cert=certificate.cert,
-        extra_flags=['-sigalgs', signature.name],
-        protocol=protocol)
+        signature_algorithm=signature,
+        protocol=protocol
+    )
+
+    if provider == GnuTLS:
+        # GnuTLS fails to validate the certificates. must run test in insecure mode.
```

Comment:
Known bug or issue?

Reply:
I get the following errors when running this without the insecure flag:
- issuer in verification was not found or insecure; trying against trust list
- Fatal error: Error in the certificate.

Could this be an issue with self signing? I switched the insecure flag with a `--no-ca-verification` flag: 10037a4b54594bb6386357420be6edd05e874c76. With this, the tests passed.

Openssl s_client has a setting that’s not enabled by default: `-verify_return_error`. From the [docs](https://www.openssl.org/docs/man1.1.1/man1/s_client.html):
`Return verification errors instead of continuing. This will typically abort the handshake with a fatal error.`

I added this flag, and the openssl client also aborted the handshake with the error: `Verify return code: 19 (self signed certificate in certificate chain)`.

Could gnutls just be more strict than openssl by default?

Reply:
Okay gnutls not liking self-signed certs makes sense, I prefer the --no-ca-verification flag since it's more specific than 'insecure'.